### PR TITLE
PR: Use control channel for comms

### LINKED ISSUE #14894: Variable explorer doesn't show variables on remote kernel when using localhost forwarded ports
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
Simply put, it's the same behavior as Issue #11538 (recently marked resolved but seems like still an issue in this case)...
I connect to an existing remote kernel, but the variable explorer never populates.

In this case, the remote kernel is accessed via localhost forwarded ports over an SSH session.
I don't know if that matters or not.

The remote system uses two factor authentication to log in even on SSH shells.
I suspect the challenge response interactivity during the login session is unexpected by the typical "remote SSH connection" configuration the Spyder console expects.  To get around this fact, or other firewall block issues, I forward the localhost ports over an existing SSH tunnel link.  Putty makes it straightforward to configure the multiple ports required in a reusable session config.

Here's an online guide:
https://medium.com/@mazzine.r/how-to-connect-your-spyder-ide-to-an-external-ipython-kernel-with-ssh-putty-tunnel-e1c679e44154

### What steps reproduce the problem?

1. Start a Kernel on the remote system using --ip=127.0.0.1 and snag the json configuration file
2. Use SSH port forwarding (e.g. PuTTY => Connection->SSH->Tunnels) to forward all remote ports in the config to localhost ports
3. Copy the JSON config locally
4. "Connect to an existing kernel" in Spyder, use the JSON config (bc of port forwards, no remote host config required)

### What is the expected output? What do you see instead?
Expected: Variable explorer to populate
Actual: Variable Explorer remains completely blank, even if variables are assigned directly inside that console window

## Versions
* Spyder version: 4.2.3 (Latest as of submission)
* Python version: 3.8.5
* Qt version: 
* PyQt version: 
* Operating System name/version: 
Local Spyder: Windows 10
Remote IPython Kernel: an updated RedHat based system I have no control over

Dependencies:
Remote IPython Kernel: 1.10.2

### PR DESCRIPTION
**Pull request by impact27**
## Description of Changes

* Removes the custom comm channel and use control channel. Uses: https://github.com/ipython/ipykernel/pull/585
* Depends on https://github.com/spyder-ide/spyder-kernels/pull/339

### Issue(s) Resolved

Fixes #14894

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

**Comment by impact27** [comment]
@ccordoba12 `from qtpy.QtWebEngineWidgets import WEBENGINE` doesn't work with qtpy 2.2.0.DEV0

**Comment by impact27** [comment]
The failed tests may not be related to this PR, see https://github.com/spyder-ide/spyder/pull/18256/checks

**Comment by ccordoba12** [comment]
> from qtpy.QtWebEngineWidgets import WEBENGINE doesn't work with qtpy 2.2.0.DEV0

Ok, that's a serious regression. @dalthviz, could you take a look at that?

> The failed tests may not be related to this PR

Yeah, @dalthviz is trying to fix them in PR #18106.

File: spyder/plugins/ipythonconsole/utils/manager.py [hunk 1]
```diff
@@ -14,10 +14,24 @@
 
 # Third party imports
 from jupyter_client.utils import run_sync
+from qtconsole.client import QtKernelClient, QtZMQSocketChannel
 import psutil
 from qtconsole.manager import QtKernelManager
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/utils/manager.py:19]:
```suggestion
import psutil
from qtconsole.client import QtKernelClient, QtZMQSocketChannel
from qtconsole.manager import QtKernelManager
```

Organize alphabetically.

File: spyder/plugins/ipythonconsole/utils/manager.py [hunk 2]
```diff
@@ -14,10 +14,24 @@
...
+    # Useful for pdb completion
+    control_channel_class = Type(QtZMQSocketChannel)
+
+
+# qtconsole and spyder-kernels depend on jupyter_client
+if hasattr(QtKernelManager, 'provisioner'):
+    JUPYTER_CLIENT_GE_7 = True
+else:
+    JUPYTER_CLIENT_GE_7 = False
+
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/utils/manager.py:34]:
```suggestion
```

This is no longer necessary.

**Review by ccordoba12 — COMMENTED** [review]
I left a couple of tiny suggestions, otherwise looks good to me.

**Comment by ccordoba12** [comment]
@impact27, please resync our subrepo with its latest `master` so we can merge this one.

File: spyder/plugins/ipythonconsole/utils/manager.py [hunk 3]
```diff
@@ -15,7 +15,15 @@
 # Third party imports
 from jupyter_client.utils import run_sync
 import psutil
+from qtconsole.client import QtKernelClient, QtZMQSocketChannel
 from qtconsole.manager import QtKernelManager
+from traitlets import Type, DottedObjectName
+
+
+class SpyderKernelClient(QtKernelClient):
+    # Enable recieving messages on control channel
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/utils/manager.py:24]:
```suggestion
    # Enable receiving messages on control channel.
```

File: spyder/plugins/ipythonconsole/utils/manager.py [hunk 4]
```diff
@@ -15,7 +15,15 @@
...
 import psutil
+from qtconsole.client import QtKernelClient, QtZMQSocketChannel
 from qtconsole.manager import QtKernelManager
+from traitlets import Type, DottedObjectName
+
+
+class SpyderKernelClient(QtKernelClient):
+    # Enable recieving messages on control channel
+    # Useful for pdb completion
+    control_channel_class = Type(QtZMQSocketChannel)
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/utils/manager.py:26]: Please move this class to `spyder/plugins/ipythonconsole/utils/client.py`. Even if it's really small, I think it's useful to have separate files for our subclasses of `QtKernelClient` and `QtKernelManager`.

**Review by ccordoba12 — COMMENTED** [review]
@impact27, a couple of small suggestions for you, then this should be ready.

**Review by ccordoba12 — APPROVED** [review]
Looks good to me, thanks @impact27!

**Pull request merged**